File: 020_retirement_py/retirement.py
working_info = dict(months = int(489), contribution = float(1000), rate_of_return = float(0.045/12))
retired_info = dict(months = int(384), contribution = float(-4000), rate_of_return = float(0.01/12))
start_age = 327 // 12
initial_savings = 21345

def retirement(start_age, initial_savings, working_info, retired_info):
    """
    Prints the current status of an individual's retirement account.
    The dictionaries both have these fields:
        "months","contribution","rate_of_return"
    Args:
       start_age (int): At what age (in months) does the individual start
       initial_savings (float): initial savings in dollars
       working_info (dict): information about working
       retired_into (dict): information about retirement
    Returns:
    None
    """
    m1 = 0
    for m1 in range(working_info['months'] + 1):
        age = (327 + m1) // 12
        m3 = (m1+3) % 12
        print("Age {:3d} month {:2d} you have ${:,.2f}".format(age, m3, initial_savings))
        if m1 == 489:
            break
        a = initial_savings * (working_info['rate_of_return'] + 1) + working_info['contribution']
        initial_savings = a
        m1 += 1
        
        
    m2 = 0
    for m2 in range(retired_info['months'] - 1):
        age = (817 + m2) // 12
        m4 = (m2+817) % 12
        b = initial_savings * (retired_info['rate_of_return'] + 1) + retired_info['contribution'] 
        initial_savings = b
        print("Age {:3d} month {:2d} you have ${:,.2f}".format(age, m4, initial_savings))
        m2 += 1    
        
retirement(start_age, initial_savings, working_info, retired_info)

File: 054_retirement/retirement.c
#include <stdio.h>
#include <stdlib.h>

struct _retire_info {
  int months;
  double contribution;
  double rate_of_return;
};
typedef struct _retire_info retire_info;

double countBalance (double balance, retire_info a) {
    balance = balance * (1 + a.rate_of_return) + a.contribution;
    return balance;
}

void retirement (int startAge, double initial, retire_info working, retire_info retired) {
    double balance = initial;
    for (int i = 0; i < working.months + 1; i++) {
        int age = (startAge + i) / 12;
        int month = (startAge + i) % 12;
        printf("Age %3d month %2d you have $%.2f\n", age, month, balance);//perhaps not printed out in format
        if (i == working.months) {
            break; //no adding to 816th month, stay where you are
        }
        balance = countBalance(balance, working);
    }
    for (int j = 1; j < retired.months ; j++) {
        int age = (startAge + working.months + j) / 12;
        int month = (startAge + working.months + j) % 12;
        balance = countBalance(balance, retired);
        printf("Age %3d month %2d you have $%.2f\n", age, month, balance);
    }
}

int main(void) {
    int startAge = 327;
    double initial = 21345;
    retire_info working = {
    489, 1000.0, 0.045 / 12
  };
    retire_info retired = {
    384, -4000.0, 0.01 / 12
  };  
    retirement (startAge, initial, working, retired);
    return 0;
}
